src/lib/features/feature-toggle/feature-toggle-service.ts

```diff
@@ -624,6 +645,11 @@ class FeatureToggleService {
             strategyConfig.variants = fixedVariants;
         }
 
+        await this.validateStrategyLimit(
+            { featureName, projectId, environment },
+            30,
```

Comment:
this is a hard limit. I would prefer the ability to configure the limit, even though we advice against it.

Reply:
yup, that's the long term plan. I found IUnleashConfig['resourceLimits'] but want to do it in another step

Reply:
I find it jumping the gun a bit. I think there is value in going via soft-limits that can be adjusted. We have limited insights into self-hosted unleash instances on on this resource.

Reply:
Can you elaborate?
My plan for the next PR was to add env var here: https://github.com/Unleash/unleash/blob/main/src/lib/create-config.ts#L625 so that you can adjust it. So the hard limit is configurable.
In addition we gonna have a soft limit when you're approaching the value that will be communicated either via toast or dedicated UI component telling you how close you are to the hard limit.

Reply:
This looks fine to me. Everything is flagged and we are all aligned that we will not ship anything that can not be configured.